### DISCUSSION TITLE
subdomain auto support

### DISCUSSION THREAD

**Discussion by wanghaisheng** [eliasdabbas/advertools#197] [Q&A] [answered]
in doc  it say i need manually pass subdomain to sitemap_todf

**Comment by eliasdabbas** [comment]
It's not clear what your question is. Please include full details.

**Comment by wanghaisheng** [comment]
now 
```


adv.crawl('https://example.com',

          'example_crawl_1.jl',

          follow_links=True

          allowed_domains=['help.example.com', 'example.com', 'community.example.com'])
```
may be
```


adv.crawl('https://example.com',

          'example_crawl_1.jl',

          follow_links=True

          allowed_sub_domains=True)
```

**Reply by eliasdabbas** [reply]
[Crawling all sub-domains](https://advertools.readthedocs.io/en/master/advertools.spider.html) is allowed by default already.




<img width="791" alt="Screen Shot 2022-04-15 at 6 52 45 PM" src="https://user-images.githubusercontent.com/15708966/163592374-35cfe5de-2db5-452b-b375-01b4ab22315d.png">


Your first comment mentions sitemap_to_df and now you are talking about crawl?

You have to communicate clearly and explain properly what you want. Take time to think what you want and double check what you wrote. I don't have time to try to figure this out for you.

**Comment by wanghaisheng** [comment, marked as answer]
that is my mistake
right now i use sitemap_to_df  at first then I wanto use crawl as plan b 
did not aware that
>[Crawling all sub-domains](https://advertools.readthedocs.io/en/master/advertools.spider.html) is allowed by default already.
